Automattic/vip-cli
# Import SQL: Add and clean up progress logs

## Description

Our logs for imports are quite sparse right now.
We've gotten requests to separate/better organize logs, especially for imports that complete really fast.

Current Logs|
|---|
|<img width="1280" alt="logs main cov" src="https://user-images.githubusercontent.com/29221840/104437902-35949880-5544-11eb-878d-1305a37f08f5.png"><img width="1278" alt="logs main 2" src="https://user-images.githubusercontent.com/29221840/104438518-f0bd3180-5544-11eb-8289-8ba0e062e071.png">|

- Adding logs for search and replace progress
- Require confirmation from users to verify search-replace values + send Track event if user bails
- Note the start & stop of each piece for better clarity (S-R, SQL validation, etc.)
- When we exit on error, log a message instead of just exiting suddenly

Also some fixes:
- Upon cancelling the confirmation prompt when the in-place flag is set, exit instead of continuing on
- Ensure we're returning the `outputFileName` when the in-place flag is set

## Steps to Test

1. Check out PR.
1. Run `npm run build`
1. Run an import:
```
VIP_PROXY="<proxy>" node ./dist/bin/vip-import-sql @2909.develop /path/to/file/dump.sql --search-replace="platform-import-test.go-vip.net,test-import.com"
```

4. Verify you see the correct progress logs:

Revised Logs|
|---|
|<img width="1280" alt="import logs 1 cov" src="https://user-images.githubusercontent.com/29221840/104436519-90c58b80-5542-11eb-870a-3d09ca43fb2b.png"><img width="1279" alt="import logs 2" src="https://user-images.githubusercontent.com/29221840/104438655-10ecf080-5545-11eb-8103-7f75a583cc77.png">|

5. Try the command again and answer `no/false` to some of the prompts. Verify that the program exits
6. This time, try the command with the `--in-place` flag and verify the appropriate logs
